configalchemy.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python
+# coding: utf8
+from __future__ import unicode_literals
+
+import sys
+import argparse
+import itertools
+
+from configparser import (BasicInterpolation, DuplicateOptionError,
+                          DuplicateSectionError, SectionProxy,
+                          MissingSectionHeaderError)
+
+from clldutils.inifile import INI as ClldIni
+
+
+class BasicReadInterpolation (BasicInterpolation):
+    """Interpolation as implemented in the classic ConfigParser.
+
+    The option values can contain format strings which refer to other values in
+    the same section or the parser's default section.
+
+    For example:
+
+        something: %(dir)s/whatever
+
+    would resolve the "%(dir)s" to the value of dir.  All reference
+    expansions are done early, at read time. If a user needs to use a
+    bare % in a configuration file, she can escape it by writing
+    %%. Other % usage is considered a user error and raises
+    `InterpolationSyntaxError'.
+
+    This class, together with INI, handles multiline interpolation
+    line by line, so
+
+        val: line1
+          line2
+          %(val)s
+          line4
+
+    would be expanded at read-time of line 3, therefore to
+
+        val: line1
+          line2
+          line1
+          line2
```

Comment:
Just reading this now. I don't know what the status of this PR is/should be. But I don't find this kind of interpolation really transparent. It means that you can change sets by re-ordering the lines in the config.

Reply:
I have to admit I was never *super* enthusiastic about this idea.  I really like the way BEASTling currently allows you to "stack" different configuration files, with rightmost files overwriting leftmost ones.  It's quick and easy and intuitive, at least IMHO.

Reply:
The reason I wanted something like this is to be able to write

    [admin]
    basename = %(basename)_with_grammar
    [model grammar]
    data = grammar.json

and the behaviour given in the patch is the one that allows this behaviour and generalizes it in the way I found obvious.

Reply:
So, independently of how we handle multi-line entries, @lmaurits this was meant to make the stacking nicer, not less nice.